packages/enzyme/src/ShallowWrapper.js

```diff
@@ -385,7 +385,9 @@ class ShallowWrapper {
             return shouldRender;
           };
         }
-        instance.setState(state, callback);
+        // We don't pass the setState callback here
+        // to guarantee to call the callback after finishing the render
```

Comment:
can we add a test case for this? or does `expect(wrapper.find('div').prop('className')).to.eql('bar');` cover that

Reply:
Yes, `expect(wrapper.find('div').prop('className')).to.eql('bar');` is a test for this.